pkg/auth/models/payload.go

```diff
@@ -3,8 +3,33 @@ package models
 //Payload is information caried by a Inspr acceess token
 type Payload struct {
 	UID        string   `json:"uid"`
-	Role       int      `json:"role"`
+	Permission []string `json:"permission"`
 	Scope      []string `json:"scope"`
 	Refresh    []byte   `json:"refresh"`
 	RefreshURL string   `json:"refreshurl"`
 }
+
+// All Permission possible values
+const (
+	CreateDapp        string = "create:dapp"
+	CreateChannel     string = "create:channel"
+	CreateChannelType string = "create:ctype"
+	CreateAlias       string = "create:alias"
+
+	GetDapp        string = "get:dapp"
+	GetChannel     string = "get:channel"
+	GetChannelType string = "get:ctype"
+	GetAlias       string = "get:alias"
+
+	UpdateDapp        string = "update:dapp"
+	UpdateChannel     string = "update:channel"
+	UpdateChannelType string = "update:ctype"
+	UpdateAlias       string = "update:alias"
+
+	DeleteDapp        string = "delete:dapp"
+	DeleteChannel     string = "delete:channel"
+	DeleteChannelType string = "delete:ctype"
+	DeleteAlias       string = "delete:alias"
+
+	CreateToken string = "create:token"
+)
```

Comment:
I actually changed on this, i noticed that in your branch there is a whole new package for the definitions of types and constants that are being used in other places of the project.

This somewhat causes confusion since it is only refered to models in other places.

Considering that these three files are not really that big, they could be store in a single file inside the `auth` pkg, allowing for a more intuitive use of the auth constants in other places.

I know that the project has other places that use the `models` scheme for separating the definitions from the implementation but the more i think about it the less that i makes sense. I even checked the kubernetes project to see if they do something similar but i couldn't find it.

At the end of the day it this follows what is found in other parts of the repo, but could be an example of another way to implement it

Reply:
Agreed